core/server/master/src/main/java/alluxio/master/metrics/TimeSeriesStore.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.metrics;
+
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * An in-memory time series store for Alluxio metrics.
+ */
+@ThreadSafe
+public class TimeSeriesStore {
+  /** <MetricName, Map<Epoch Time, Value>>. **/
+  private final Map<String, TreeMap<Long, Long>> mTimeSeries;
```

Comment:
Would `List<DataPoint>` be a more natural type to use here? I don't see a need for map-style access

Reply:
Sure, the serialized json would come out the same though?